apps/omg_api/test/state/core_test.exs

```diff
@@ -748,6 +775,54 @@ defmodule OMG.API.State.CoreTest do
     |> fail?(:utxo_not_found)
   end
 
+  @tag fixtures: [:alice, :bob, :state_empty]
+  test "can spend utxos with mixed currencies", %{
+    alice: alice,
+    bob: bob,
+    state_empty: state
+  } do
+    state
+    |> Test.do_deposit(alice, %{amount: 1, currency: eth(), blknum: 1})
+    |> Test.do_deposit(alice, %{amount: 2, currency: not_eth(), blknum: 2})
+    |> (&Core.exec(
+          Test.create_recovered([{1, 0, 0, alice}, {2, 0, 0, alice}], [{bob, eth(), 1}, {bob, not_eth(), 2}]),
+          zero_fees_map(),
+          &1
+        )).()
+    |> success?
+  end
+
+  @tag fixtures: [:alice, :bob, :state_empty]
+  test "respects fees for transactions with mixed currencies", %{
```

Comment:
ah, there's the fee-test, so the behavior is not like the suggested (fee is covered in at least one currency).

Make sure that you test also situations like when fee is covered in `eth()` but `not_eth()` leg of the tx doesn't even have enough to cover the output, to ensure that fee handling doesn't surprisingly break the "basic" validity.